headers.go

```diff
@@ -0,0 +1,40 @@
+package goka
```

Comment:
If breaking changes are accepted, I'd like to create `type Headers map[string][]byte` and define the functions below with that type as the receiver. It would make the code base more readable and future enhancements more flexible.

Otherwise, maybe just an alias for the readability part: `type Headers = map[string][]byte` which will keep everything backwards compatible and the change would be just a rename.

Reply:
To be honest I don't mind the breaking change, especially since it will be obvious after an upgrade and can be fixed with simple replacement.
What's your opinion on that @jomaresch @R053NR07  @mmreza79?

Reply:
I agree. ✅
Having `type Headers map[string][]byte` will improve the readability. 
I also don't mind breaking changes.

Reply:
awesome, @mrahs let's do it

Reply:
Thanks. I'll submit another PR for that change.

Reply:
Ah.. looks like I ended up updating this PR.